promql/engine.go

```diff
@@ -336,77 +285,64 @@ func (ng *Engine) NewRangeQuery(qs string, start, end clientmodel.Timestamp, int
 		Interval: interval,
 	}
 
-	query := &query{
+	qry := &query{
 		q:     qs,
 		stmts: Statements{es},
 		ng:    ng,
-		done:  make(chan bool, 2),
 		stats: stats.NewTimerGroup(),
 	}
-	return query, nil
+	return qry, nil
 }
 
-// exec executes all statements in the query. For evaluation statements only
-// one statement per query is allowed, after which the execution returns.
+// testStmt is an internal helper statement that allows execution
+// of an arbitrary function during handling. It is used to test the Engine.
```

Comment:
Arguably this should be moved to a `..._test.go` file if it's only used in tests?